lib/web_ui/dev/felt

```diff
@@ -57,8 +57,10 @@ KERNEL_NAME=`uname`
 if [[ $KERNEL_NAME == *"Darwin"* ]]
 then
   echo "Running on MacOS. Increase the user limits"
-  ulimit -n 50000
-  ulimit -u 4096
+  ULIMIT_FILES=`ulimit -n`
+  ulimit -n $ULIMIT_FILES
```

Comment:
Isn't this setting the ulimit to the same number that is currently set?

Reply:
Ah one second I thought it will show the maximum, let me take look :)

Reply:
In my mac it shows:

```
$ ulimit -n
256
```

And IIRC the max is `1024`.

Reply:
Actually my ulimit -n is... a little bit odd:

```
$ ulimit -n 1045
$ ulimit -n 50000
-bash: ulimit: open files: cannot modify limit: Operation not permitted
$ ulimit -n 1024
$ ulimit -n 2048
-bash: ulimit: open files: cannot modify limit: Operation not permitted
$ ulimit -n 1512
-bash: ulimit: open files: cannot modify limit: Operation not permitted
$ ulimit -n 1100
-bash: ulimit: open files: cannot modify limit: Operation not permitted
$ ulimit -n 1080
-bash: ulimit: open files: cannot modify limit: Operation not permitted
$ ulimit -n 1040
-bash: ulimit: open files: cannot modify limit: Operation not permitted
$ ulimit -n 1020
$ ulimit -n 1024
-bash: ulimit: open files: cannot modify limit: Operation not permitted
$ ulimit -n 1024
-bash: ulimit: open files: cannot modify limit: Operation not permitted
```

So I don't know what to suggest on how to find the actual max :S

Reply:
I found the solution :) I'll update the PR tomorrow, however once you start getting that error in MacOS, you need to restart.

I'll first check the maxuser limit, increase it if necessary, later set the user limit. Also I'll only set the ulimit if it's too low, so I won't try to change it again and again.

Reply:
Should we also check the exit code from `ulimit`? If it's 0, then everything is great, moving on. If it's non-zero, print a big warning banner but still continue running the test. It's not always possible for the user to perform `sudo` stuff. Also, not all `felt` invocations require high `ulimit`. For example, you should still be able to run one unit test, which may be enough.

Reply:
@ditman please have another look :)